Thanks for the PR. Could you please link issues fixed by this PR? (Adding `Fixes #1234` in the description)

Also, obviously, CI is failing :) Could you have a look? It look like a Java Service needs to be registered:

```
[INFO] 	
[INFO] -------------------------------------------------------	
[INFO]  T E S T S	
[INFO] -------------------------------------------------------	
[INFO] Running io.quarkus.it.hibernate.reactive.postgresql.HibernateReactiveFetchLazyInGraalIT	
Executing "/home/runner/_work/quarkus/quarkus/integration-tests/hibernate-reactive-postgresql/target/quarkus-integration-test-hibernate-reactive-postgresql-999-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/home/runner/_work/quarkus/quarkus/integration-tests/hibernate-reactive-postgresql/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus.profile=prod -Dquarkus.native.container-build=true -Dquarkus.native.native-image-xmx=6g"	
Apr 08, 2025 4:31:16 PM io.quarkus.runtime.ApplicationLifecycleManager run	
ERROR: Failed to start application	
java.lang.RuntimeException: Failed to start quarkus	
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)	
	at io.quarkus.runtime.Application.start(Application.java:101)	
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)	
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:80)	
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:51)	
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:144)	
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)	
Caused by: java.util.ServiceConfigurationError: io.vertx.core.spi.VertxServiceProvider: Provider org.hibernate.reactive.context.impl.ContextualDataStorage not found	
	at java.base@21.0.6/java.util.ServiceLoader.fail(ServiceLoader.java:593)	
	at java.base@21.0.6/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1219)	
	at java.base@21.0.6/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1228)	
	at java.base@21.0.6/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273)	
	at java.base@21.0.6/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309)	
	at java.base@21.0.6/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393)	
	at io.vertx.core.ServiceHelper.loadFactories(ServiceHelper.java:57)	
	at io.vertx.core.ServiceHelper.loadFactories(ServiceHelper.java:44)	
	at io.vertx.core.impl.VertxBuilder.init(VertxBuilder.java:274)	
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder.initialize(VertxCoreRecorder.java:245)	
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$VertxSupplier.get(VertxCoreRecorder.java:698)	
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$VertxSupplier.get(VertxCoreRecorder.java:677)	
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.initializeRouter(VertxHttpRecorder.java:324)	
	at io.quarkus.runner.recorded.VertxHttpProcessor$preinitializeRouter1141331088.deploy_0(Unknown Source)	
	at io.quarkus.runner.recorded.VertxHttpProcessor$preinitializeRouter1141331088.deploy(Unknown Source)	
	... 7 more
```

Thanks.

--- a/extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java
+++ b/extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java
@@ -57,6 +57,7 @@ import io.quarkus.deployment.builditem.LogCategoryBuildItem;
 import io.quarkus.deployment.builditem.SystemPropertyBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
 import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
 import io.quarkus.deployment.recording.RecorderContext;
 import io.quarkus.hibernate.orm.deployment.HibernateConfigUtil;
@@ -94,6 +95,13 @@ public final class HibernateReactiveProcessor {
             "org.hibernate.reactive.persister.collection.impl.ReactiveOneToManyPersister",
             "org.hibernate.reactive.persister.collection.impl.ReactiveBasicCollectionPersister",
     };
+
+    @BuildStep
+    void registerServicesForReflection(BuildProducer<ServiceProviderBuildItem> services) {
+        services.produce(new ServiceProviderBuildItem(
+                "io.vertx.core.spi.VertxServiceProvider",
+                "org.hibernate.reactive.context.impl.ContextualDataStorage"));
+    }
 
     @BuildStep
     void registerBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans, CombinedIndexBuildItem combinedIndex,

--- a/pom.xml
+++ b/pom.xml
@@ -75,7 +75,7 @@
         <antlr.version>4.13.0</antlr.version> <!-- version controlled by Hibernate ORM's needs -->
         <bytebuddy.version>1.15.11</bytebuddy.version> <!-- version controlled by Hibernate ORM's needs -->
         <hibernate-commons-annotations.version>7.0.3.Final</hibernate-commons-annotations.version> <!-- version controlled by Hibernate ORM's needs -->
-        <hibernate-reactive.version>2.4.5.Final</hibernate-reactive.version> <!-- highly sensitive to Hibernate ORM upgrades -->
+        <hibernate-reactive.version>2.4.6.Final</hibernate-reactive.version> <!-- highly sensitive to Hibernate ORM upgrades -->
         <hibernate-validator.version>8.0.2.Final</hibernate-validator.version>
         <hibernate-search.version>7.2.3.Final</hibernate-search.version>
 

